event.py

```diff
@@ -0,0 +1,55 @@
+from copy import deepcopy
+from datetime import datetime, time
+from flask_restful import abort
+
+from eventdao import EventDAO
+
+def read_time(time_str):
+    return datetime.strptime(time_str, '%H:%M:%S').time()
+
+class Event:
+    dao = EventDAO()
+
+    def __init__(self, document: dict):
+        self.id = str(document['_id']) if '_id' in document else None
+        self.name = document['name'] if 'name' in document else None
+        self.days = document['days'] if 'days' in document else None
+        self.from_time = document['from_time'] if 'from_time' in document else None
+        self.to_time = document['to_time'] if 'to_time' in document else None
+        self.description = document['description'] if 'description' in document else None
+
+    @classmethod
+    def get_by_name(cls, event_name):
+        event_dict = cls.dao.get_by_name(event_name)
+        if event_dict is None:
+            abort(404, message='Event {} not found'.format(event_name))
+        return cls(event_dict)
+
+    @classmethod
+    def get_by_id(cls, id_str):
+        event_dict = cls.dao.get_by_id(id_str)
+        if event_dict is None:
+            abort(404, message='Event with id "{}" not found'.format(id_str))
+        return cls(event_dict)
+
+    @classmethod
+    def get_all(cls):
+        event_dicts = cls.dao.get_all()
+        return [cls(e) for e in event_dicts]
```

Comment:
are you sure this returns what you are expecting it to return ?

Reply:
I thought it returns a list of event objects...